app/models/user.rb

```diff
@@ -55,8 +55,10 @@ class User < ActiveRecord::Base
         where(["#{self.table_name}.admin = ? OR #{Usergroup.table_name}.admin = ?", true, true])
   }
   scope :except_hidden, lambda {
-    if (hidden = AuthSourceHidden.all).present?
-      where("#{self.table_name}.auth_source_id <> ?", hidden)
+    if (hidden = AuthSourceHidden.where(nil).to_a).present?
+      where(:"#{self.table_name}.auth_source_id" => AuthSource.all - hidden)
```

Comment:
It looks like this will introduce a second SQL query for the AuthSource.all - so we'll now have one to retrieve the hidden ID(s), another to retrieve all AuthSource IDs, and then finally the user query.  Can we avoid the extra query?